model/ddcuri/ddcuri.go

```diff
@@ -0,0 +1,50 @@
+package ddcuri
+
+import (
+	"fmt"
+	"strings"
+)
+
+type DdcQuery struct {
+	Organization string
+
+	BucketName  string
+	BucketId    uint32
+	BucketIdSet bool
+
+	Protocol string
+	Cid      string
+	Path     []string
+
+	Options string
+}
+
+func Parse(uri string) (q DdcQuery, err error) {
+	uri = consumeOptions(&q, uri)
+	err = consumeMain(&q, uri)
+	return q, err
+}
+
+func ParseWebUrl(url string) (q DdcQuery, err error) {
```

Comment:
this method is going to always fail. Here you cutting off 'ddc' prefix (constant could be used). But inside  Parse method you invoke 'consumeMain' that fails in case 'ddc' prefix isn't presented

Reply:
unit test could help here